### PR TITLE
Feat/various 2

### PR DESCRIPTION
**Pull request by streamich**

### CODE REVIEW

File: docs/BatterySensor.md [hunk 1]
```diff
@@ -26,3 +26,39 @@ Result
...
+or merge the props
+
+```js
+const MyCompWithBattery = withBattery(MyComp, null);
+```
+
+
+## `@withBattery`
+
+Statefull component class decorator that injects `battery` prop into your component.
```
Comment (xram) [inline docs/BatterySensor.md]: `Stateful` has on `l`.

File: docs/MediaDeviceSensor.md [hunk 2]
```diff
@@ -6,9 +6,34 @@ user's connected media devices.
...
+```js
+import {withMediaDevices} from 'libreact/lib/MediaDeviceSensor';
+
+const MyCompDevices = withMediaDevices(MyComp);
+```
+
+
+## `@withMediaDevices`
+
+Statefull component class decorator that injects `devices` prop into your component.
```
Comment (xram) [inline docs/MediaDeviceSensor.md]: Here again.

File: docs/MediaSensor.md [hunk 3]
```diff
@@ -11,3 +11,28 @@ import {MediaSensor} from 'libreact/lib/MediaSensor';
...
+```js
+import {withMedia} from 'libreact/lib/MediaSensor';
+
+const MyCompWithSize = withMedia(MyComp, 'isBigScreen', '(min-width: 480px)');
+```
+
+
+## `@withMedia`
+
+Statefull component class decorator that injects a boolean prop into your component that specifies if media query matches.
```
Comment (xram) [inline docs/MediaSensor.md]: Maybe just do a find and replace for these :-)

File: docs/Video.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+}
+```
+
+  - `src` - required, string, video source file URL.
+  - `autoPlay` - optional, boolean, whether to autoplay media, defaults to `false`.
+  - `loop` - optional, boolean, whether to repeat the media when it ends, defaults to `false`.
+  - `muted` - optional, boolean, whether to mute the media, defaults to `false`.
+  - `preload` - optional, string, `<video>` element preload attribute.
+  - `volume` - optional, number, media volume in `[0..1]` range, defaults to `1`.
+  - `noJs` - optional, React element(s) to render insided the `<video>` tag.
```
Comment (xram) [inline docs/Video.md]: Please check the syntax of this sentence.

File: src/Media/index.ts [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+  componentDidMount () {
+    if (this.props.autoPlay && this.el.paused) {
+      this.play();
+    }
+
+    this.setState({
+      volume: this.el.volume
+    });
+
+    this.event('onMount')(this);
```
Comment (xram) [inline src/Media/index.ts]: Is `onMount` the event name here?

File: src/Media/index.ts [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+  componentWillUnmount () {
+    this.el = null;
+
+    this.event('onUnmount')(this);
+  }
+
+  play = () => {
+    if (this.el) {
+      // TODO: In some browsers `.play()` method returns a `Promise`, where you
```
Comment (xram) [inline src/Media/index.ts]: Should this TODO be done already?

File: src/Media/index.ts (hunk 5 above)
Comment (streamich) [reply src/Media/index.ts]: Yes

File: docs/Video.md (hunk 4 above)
Comment (streamich) [reply docs/Video.md]: Looks good to me.

File: docs/BatterySensor.md (hunk 1 above)
Comment (streamich) [reply docs/BatterySensor.md]: Done.

File: docs/MediaDeviceSensor.md (hunk 2 above)
Comment (streamich) [reply docs/MediaDeviceSensor.md]: Done.

File: docs/MediaSensor.md (hunk 3 above)
Comment (streamich) [reply docs/MediaSensor.md]: Done.

**Pull request merged**